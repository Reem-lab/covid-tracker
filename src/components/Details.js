/* eslint-disable no-underscore-dangle */
import { BsArrowRightCircle } from 'react-icons/bs';
import { FaVirus } from 'react-icons/fa';
import { useSelector, useDispatch } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { displayInformations } from '../redux/actions/AllInfo';
import '../styles/Details.css';

const Details = () => {
  const params = useParams();
  const countriesInfo = useSelector((state) => state.RegionsReducer);
  const filtered = countriesInfo.filter((item) => item.country === params.country);
  const dispatch = useDispatch();

  useEffect(() => {
    if (countriesInfo.length === 0) {
      dispatch(displayInformations());
    }
  }, []);

  return (
    <div className="dRegions">
      { !filtered.length ? (
        <div className="heading">No Counrties Information and Details Found ❗❕</div>
      ) : (filtered.map((info) => (
        <div key={info.country}>
          <div className="header">
            <FaVirus className="virsus-icon" />
            <div className="naCases">
              <h1>
                {info.country}
                {' '}
              </h1>
              <span>{info.cases}</span>
            </div>
          </div>
          <div>
            <div className="info">
              <p>Today cases in the country</p>
              <div className="cases-number">
                <span>{info.todayCases}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>General deaths</p>
              <div className="cases-number">
                <span>{info.deaths}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>Total Recovered</p>
              <div className="cases-number">
                <span>{info.recovered}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>Continent name</p>
              <div className="cases-number">
                <span>{info.continent}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>Total tests that made</p>
              <div className="cases-number">
                <span>{info.tests}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>Today deaths</p>
              <div className="cases-number">
                <span>{info.todayDeaths}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
            <div className="info">
              <p>Today Recovered</p>
              <div className="cases-number">
                <span>{info.todayRecovered}</span>
                <Link to="/"><BsArrowRightCircle className="arrow" /></Link>
              </div>
            </div>
          </div>
        </div>
      )))}
    </div>
  );
};

export default Details;
