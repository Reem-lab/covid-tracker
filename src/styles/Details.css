
.info {
  display: flex;
  background-color: #35548b;
  color: #fff;
  justify-content: space-between;
  align-items: baseline;
  padding: 15px;
  font-size: 20px;
  height: 80px;
}

.info:nth-child(2n) {
  background-color: #5485e1;
}

.cases-number {
  display: flex;
  gap: 20px;
}

.header {
  display: flex;
  justify-content: space-between;
  background-color: #5485e1;
  padding: 15px;
  color: #fff;
  height: 110px;
}

.arrow {
  cursor: pointer;
  color: #fff;
}

.naCases {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

@media screen and (max-width: 768px) {
  .cases-number {
    gap: 4px;
  }

  .info {
    font-size: 18px;
  }
}
