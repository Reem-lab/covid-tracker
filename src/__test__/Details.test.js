/* eslint-disable no-unused-vars */
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Details from '../components/Details';
import store from '../redux/store';

describe('Test the Countries list', () => {
  it('renders correctly', () => {
    const rocket = [
      {
        country: 'Afghanstan',
        cases: 222,
      },
    ];

    const { container } = render(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
    const message = screen.getByText('No Counrties Information and Details Found ❗❕');
    expect(message).toMatchSnapshot();
  });
});
