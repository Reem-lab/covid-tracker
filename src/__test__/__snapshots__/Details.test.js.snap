// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Countries list renders correctly 1`] = `
<div
  class="heading"
>
  No Counrties Information and Details Found ❗❕
</div>
`;
